usautobuild/logger.py

```diff
@@ -1,39 +1,127 @@
-import logging
-from logging import handlers
 import datetime
 import sys
+import random
+import re
+import logging
+from logging import handlers
+from pathlib import Path
+
+import requests
+
+from .config import Config
+
+log = logging.getLogger("usautobuild")
+
+
+def str_to_log_level(s: str) -> int:
+    level_name_map = {
+        "DEBUG": logging.DEBUG,
+        "INFO": logging.INFO,
+        "WARNING": logging.WARNING,
+        "ERROR": logging.ERROR,
+    }
+
+    return level_name_map.get(s.upper(), logging.INFO)
+
 
 def setup_logger(arg_level: str):
+    """Configure basic logging facilities"""
+
+    log.setLevel(str_to_log_level(arg_level))
+
+    log_path = Path("logs")
+    log_path.mkdir(exist_ok=True)
+
+    fmt = logging.Formatter("[%(asctime)s::%(name)s::%(levelname)s] %(message)s")
+
+    for handler in (
```

Comment:
I don't think this change made it particularly clearer to read. I get it is able to have more handlers now, but will we ever?

Reply:
rolled back